batch7crud-roll10/src/main/java/com/lftechnology/batch7crud/utils/requestmapper/RequestMapper.java

```diff
@@ -0,0 +1,213 @@
+    }
+  }
+
+
+  private Class filterController(List<Class> controllers, String servletPath){
+    List<String> servletPathParts = new ArrayList<>();
+    String[] pathPartsAsArray = servletPath.split("/");
+    for(String part : pathPartsAsArray){
+      if(part != null && !part.isEmpty() && !part.startsWith("*")){
+        servletPathParts.add(part);
+      }
+    }
+
+    for(Class controller : controllers){
+      if(controller.isAnnotationPresent(WebServlet.class)){
+        WebServlet webServlet = (WebServlet) controller.getAnnotation(WebServlet.class);
+
+        List<String> pathValueParts = new ArrayList<>();
+        String[] valuePartsAsArray = webServlet.value()[0].split("/");
+
+        for(String part : valuePartsAsArray){
+          if(part != null && !part.isEmpty() && !part.startsWith("*")){
+            pathValueParts.add(part);
+          }
+        }
+
+        if(isSame(servletPathParts, pathValueParts)){
+          return controller;
+        }
+      }
+    }
+    return null;
+  }
+
+  private Boolean isSame(List<String> param1, List<String> param2){
+    if(param1.size() != param2.size())
+      return false;
+
+    for (int i = 0; i < param1.size(); i++) {
+      if(!param1.get(i).equalsIgnoreCase(param2.get(i)))
+        return false;
+    }
+
+    return true;
+
+  }
+
+  private Method filterMethod(List<Method> methodList, String pathInfo, String requestType){
+    //filter method by annotation
+    List<Method> annotatedMethods = filterMethodByAnnotation(RequestMapping.class, methodList);
+
+    //filter method by request type (GET or POST)
+    annotatedMethods = filterMethodByRequestType(annotatedMethods, requestType);
+
+    //filter method by path size
+    annotatedMethods = filterMethodByPathSize(annotatedMethods, pathInfo);
+
+    return filterRequestedMethod(pathInfo,0, annotatedMethods);
+  }
+
+  private Method filterRequestedMethod(String pathInfo, Integer index, List<Method> methods){ // NOSONAR
+    String[] pathParts = pathInfo.split("/");
+    List<String> pathParts1 = new ArrayList<>();
+    for(String p : pathParts){
+      if(p != null && !p.isEmpty()){
+        pathParts1.add(p);
+      }
+    }
+
+    List<Method> candidateMethods = new ArrayList<>();
+    for(Method method : methods){
+      RequestMapping mapping = method.getAnnotation(RequestMapping.class);
+      String value = mapping.value();
+
+      String[] valueParts = value.split("/");
+
+      /*
+      * condition 1 : if the index value is less than that of value parts length
+      * condition 2 : if the value at index is variable type
+      * condition 3 : if the value at index matches path parts at index
+      * */
+      if(index < valueParts.length && (isVariable(valueParts[index]) || valueParts[index].equalsIgnoreCase(pathParts1.get(index)))){
+        candidateMethods.add(method);
+
+      }
+    }
+
+
+    if(index < pathParts1.size() - 1){
+      return  filterRequestedMethod(pathInfo, ++index, candidateMethods);
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Introduce a new variable instead of reusing the parameter "index". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1226)